pdns/pdnsutil.cc

```diff
@@ -1204,6 +1218,28 @@ static int editZone(const DNSName &zone, const PDNSColors& col) {
     return EXIT_FAILURE;
   }
 
+  if (isatty(STDIN_FILENO) == 0) {
+    cerr << "edit-zone requires a terminal" << endl;
```

Comment:
In theory, the editor could be a script that applies patches, using sed or whatever. However, everything edit-zone does around editing still requires a terminal, so I think this check is still good.